### PR TITLE
Remove templated metadata.yaml and populate initial charmcraft yaml

### PR DESCRIPTION
**Pull request by mthaddon**
Applicable spec: N/A

### Overview

Remove templated `metadata.yaml` and populate initial `charmcraft.yaml` so that very basic charm can be built. Also update so basic lint and unit tests pass.

### Rationale

Allows us to build a charm on 24.04.

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

N/A

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

No documentation for charmhub yet, that'll come later once we have a charm that actually deploys a workload.

### CODE REVIEW

**Review by yanksyoon — APPROVED** [review]
LGTM!

**Review by weiiwang01 — APPROVED** [review]

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -15,8 +15,8 @@
 
 logger = logging.getLogger(__name__)
 
-METADATA = yaml.safe_load(Path("./metadata.yaml").read_text(encoding="utf-8"))
-APP_NAME = METADATA["name"]
+CHARMCRAFT = yaml.safe_load(Path("./charmcraft.yaml").read_text(encoding="utf-8"))
+APP_NAME = CHARMCRAFT["name"]
```
Comment (javierdelapuente) [inline tests/integration/test_charm.py:19]: just a comment, this could be anything, no need to read the charmcraft.yaml file

**Review by javierdelapuente — APPROVED** [review]
LGTM

File: tests/integration/test_charm.py (hunk 1 above)
Comment (mthaddon) [reply tests/integration/test_charm.py:19]: Yeah, true. I don't think it's a big problem but something to bear in mind, possibly for future PRs.

**Pull request merged**